### ISSUE TITLE
Training New Models

### ISSUE THREAD

**Issue by changcheng967** [Doulet-Media/Kata_speed#178]
# Training New b18c384cbt Models Specialized for 9x9

This repository contains the details and progress of training new `b18c384cbt` models specialized for 9x9 Go using the base model `kata9x9-b18c384nbt-20231025.bin.gz`.

## Overview

We are excited to announce the start of our second training session for the new `b18c384cbt` models, specifically designed for 9x9 Go. The base model `kata9x9-b18c384nbt-20231025.bin.gz` will be used as the starting point for this training.

## Training Details

- **Model Type**: `b18c384cbt`
- **Board Size**: 9x9
- **Base Model**: `kata9x9-b18c384nbt-20231025.bin.gz`
- **Training Session**: Second session, starting with 50 games

## Resolved Module Issues

We have resolved the missing module issues to ensure a smooth training process:
- **numpy**: Successfully installed
- **packaging**: Successfully installed
- **torch**: Successfully installed
- **psutil**: Successfully installed

## Training Process

1. **Setup**:
    - Ensure the environment is configured with the necessary dependencies, including CUDA and cuDNN.

2. **Running Self-play**:
    - Execute the self-play script to generate training data:
    ```bash
    ./katago selfplay -models-dir /path/to/models -config /path/to/config -output-dir /path/to/output -max-games-total 50 -override-config numGameThreads=5
    ```

3. **Data Shuffling**:
    - Shuffle the generated training data to prepare it for training:
    ```bash
    ./shuffle.sh /path/to/output /path/to/shufflescratch 8 128 -min-rows 100000 -keep-target-rows 600000 -taper-window-scale 50000
    ```

4. **Training**:
    - Train the model using the shuffled data:
    ```bash
    ./train.sh /path/to/shuffled/data /path/to/models /path/to/config
    ```

## Update: Second Training Session

Following the initial training session, we have successfully resolved the missing module issues, including `numpy`, `packaging`, `torch`, and `psutil`. With these dependencies now properly installed, we have commenced the second training session.

### Command to Start Training

1. **Navigate to the selfplay directory**:
    ```bash
    cd /home/changcheng967/changcheng967/KataGo/python/selfplay
    ```

2. **Run the `synchronous_loop.sh` script** with the required arguments:
    ```bash
    ./synchronous_loop.sh myrun /home/changcheng967/changcheng967/KataGo/python/selfplay mytraining b18c384cbt 1
    ```

### Progress

- The second training session has begun with a fixed environment, running the first batch of 50 games.
- The training data is being generated and processed as expected, and the shuffle and training scripts are executing without issues.

### Next Steps

- Monitor the training session and ensure all games complete successfully.
- Analyze the generated data and update the models accordingly.
- Prepare for subsequent training sessions to further refine and improve the models.

Thank you for your patience and support as we work on enhancing the performance of the `b18c384cbt` models for 9x9 Go. Stay tuned for further updates!

## Contact

For any questions or inquiries, please open an issue in this repository or reach out to our team.

---

Thank you for your interest and support!
